File: CaseStudies/swpcm/Program/energy3.m
%% Energy Module, when Tp > Tmelt
% This module uses the input parameters in params and the array T3 which is calculated 
% by temp3.m to specify the equations that govern the change in energy of the water and PCM
%
% Authors: Thulasi Jegatheesan, Spencer Smith and Ned Nedialkov
%
% Data Last Revised: May 20, 2015
%
% Uses: params (Input Parameters Module)
%
% State Variables: none
%
% Environment Variables: none
%
% Governing Equations
%
% $$E_W(t) = C_W m_w (T_W(t) - T_{init})$$
%
% $$E_P(t) = E^{init}_{Pmelt} + H_f m_P + C^L_P m_P (T_P(t) - T^P_{melt})$$
%
%
function [Ew3, Ep3] = energy3(T3, params)

    %change in energy in water when T > Tmelt
    Ew3     = params.C_w * params.Mw * (T3(:,1) - params.Tinit);
        
    
    %change in energy in Pcm when T > Tmelt        
    
    Ep3     = params.Epmelt_init + params.Ep_melt3 + (params.C_pl * params.Mp * (T3(:,2) - params.Tmelt));

File: CaseStudies/SolarHeatingWaterOnly/CCode/energy3.c
/* Energy Module, when Tp > Tmelt

This module uses the input parameters in params and the temperatures of
water and PCM calculated by main.c and the temperature3 function to specify
the equations that govern the change in energy of the water and PCM.

Authors: Thulasi Jegatheesan, Spencer Smith, Ned Nedialkov, and Brooks
MacLachlan

Date Last Revised: June 10, 2016

Governing Equations:

Ew(t) = Cw * Mw * (Tw(t) - Tinit)

Ep(t) = Epmelt_init + Qp(t)

*/

#include <stdio.h>
#include <stdlib.h>
#include "parameters.h"
#include "energy3.h"

double energy3Wat(double tWat, struct parameters params){

    double eWat;

    // Change in energy in water when T > Tmelt
    eWat = params.C_w * params.Mw * (tWat - params.Tinit);

    return eWat;
}

double energy3PCM(double tPCM, struct parameters params){

    double ePCM;

    // Change in energy in PCM when T > Tmelt
    ePCM = params.Epmelt_init + params.Ep_melt3 + params.C_pl * params.Mp * (tPCM - params.Tmelt);

    return ePCM;
}
